### PR TITLE
Allow handlers on different nesting layers, closes nyxx#233

### PR DESCRIPTION
**Pull request by abitofevrything**
# Description

Previously handlers could not be registered for both subcommands and sub-subcommands because of the way preventing root handlers on slash commands with subcommands was implemented. This PR reworks that system to allow both subcommands and sub-subcommands to have handlers, while still preventing slash commands with subcommands having roor handlers.

Also added .gitignore since it was missing, it is the same as [nyxx](https://github.com/nyxx-discord/nyxx/blob/dev/.gitignore).

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] Ran `dartanalyzer --options analysis_options.yaml .`
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by abitofevrything** [comment]
Also closes #3.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+/nyxx.wiki/
+/test/private.dart
+/publish_docs.sh
+/test/mirrors.dart
+/private
+private-*.dart
+test-*.dart
+[Rr]pc*
+**/doc/api/**
+old.*
```
Comment (l7ssha) [inline .gitignore:36]: no new line at the end of file

**Review by l7ssha — CHANGES_REQUESTED** [review]
One small comment otherwise looks good

**Comment by abitofevrything** [comment]
Newline has been added.

**Review by l7ssha — APPROVED** [review]
Thanks, I will marge it later this day and 2.1.1 version will be pushed also

**Pull request merged**